snippets/base/admin/actions.py

```diff
@@ -1,8 +1,216 @@
-from django.db import transaction
+import json
+
+from django.contrib import messages
+from django.db import IntegrityError, transaction
+
+from raven.contrib.django.models import client as sentry_client
+
+from snippets.base.models import (STATUS_CHOICES, Addon, ASRSnippet, Campaign,
+                                  SnippetTemplate, Target)
+from snippets.base.forms import PROFILE_AGE_CHOICES, PROFILE_AGE_CHOICES_ASR, TargetAdminForm
 
 
 @transaction.atomic
 def duplicate_snippets_action(modeladmin, request, queryset):
     for snippet in queryset:
         snippet.duplicate(request.user)
 duplicate_snippets_action.short_description = 'Duplicate selected snippets'  # noqa
+
+
+@transaction.atomic
+def migrate_snippets_action(modeladmin, request, queryset):
+    for snippet in queryset:
+        if not snippet.on_startpage_5:
+            messages.error(request, 'Only Activity Stream Snippets can be migrated.')
+            return
+
+    migrated_snippets = 0
+    skipped = 0
+    try:
+        with transaction.atomic():
```

Comment:
you've already decorated this function with `transaction.atomic`. redundant?

Reply:
Looks like it but it's not really. That's how the Django docs are suggesting to do it, to be able to handle the exceptions nicely

https://docs.djangoproject.com/en/2.1/topics/db/transactions/#controlling-transactions-explicitly